lib/jekyll/page.rb

```diff
@@ -40,7 +40,11 @@ def initialize(site, base, dir, name)
       @base = base
       @dir  = dir
       @name = name
-      @path = site.in_source_dir(base, dir, name)
+      @path = if site.in_theme_dir(base) == base # we're in a theme
```

Comment:
@benbalter When you include `assets/styles.scss` with YAML front matter, it's a `Jekyll::Page`. To ensure it's read properly, we need to make sure we're looking in the theme's directory rather than your site's directory. This `if` says "if this Page is inside our theme directory then ensure the `@path` gets set properly."